### PR TITLE
Fix Service to be an optional field. Add support for updating ExtraLabels.

### PR DESCRIPTION
**Pull request by Dean-Coakley**
### Proposed changes
* In https://github.com/nginxinc/nginx-ingress-operator/pull/58 an optional field called `Service` was added, but the generation of any KIC service depended on it existing, added a nil check to prevent panics: https://github.com/nginxinc/nginx-ingress-operator/commit/ad81ccf7ba5816da4e384e58ef9fba4c2ea9f6df
* In https://github.com/nginxinc/nginx-ingress-operator/pull/58 I added support for adding extraLabels, however I missed the case of updating an already existing service. That is added in: https://github.com/nginxinc/nginx-ingress-operator/commit/38152435eee6d20f4240741d70da8541de28c73e

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-ingress-operator/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: pkg/controller/nginxingresscontroller/service_test.go [hunk 1]
```diff
@@ -66,3 +65,256 @@ func TestServiceForNginxIngressController(t *testing.T) {
...
+			"override labels",
+		},
+		{
+			&corev1.Service{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+				Spec: corev1.ServiceSpec{
+					Type: corev1.ServiceTypeLoadBalancer,
```
Comment (pleshakov) [inline pkg/controller/nginxingresscontroller/service_test.go:290]: there are no labels here. Are we really removing them?

File: pkg/controller/nginxingresscontroller/service_test.go [hunk 2]
```diff
@@ -66,3 +65,256 @@ func TestServiceForNginxIngressController(t *testing.T) {
...
+				ObjectMeta: v1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+					Labels:    extraLabels,
+				},
+				Spec: corev1.ServiceSpec{
+					Type: corev1.ServiceTypeLoadBalancer,
+				},
+			},
+			"override labels",
```
Comment (pleshakov) [inline pkg/controller/nginxingresscontroller/service_test.go:281]: looks like we're adding labels, but not overriding

File: pkg/controller/nginxingresscontroller/service_test.go [hunk 3]
```diff
@@ -66,3 +65,256 @@ func TestServiceForNginxIngressController(t *testing.T) {
...
+			},
+			true,
+			"remove label",
+		},
+		{
+			&corev1.Service{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+					Labels:    extraLabels,
```
[the remark this answers is not in the document]
Comment (Dean-Coakley) [reply pkg/controller/nginxingresscontroller/service_test.go:181]: I'm removing the `Service` struct from the CRD(desired state), and testing that labels are removed from the service if it was added from a previous version of the CRD.

Based on this comment I considered adding `including labels` to the test case message but I think that note will be likely misleading as soon as we extend the `Service` struct.

**Review by pleshakov — APPROVED** [review]

**Review by soneillf5 — APPROVED** [review]
:shipit:

**Pull request merged**